instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/decorators/Http4SpanDecorator.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.apachecamel.decorators;
+
+import javax.annotation.Nullable;
+import org.apache.camel.Endpoint;
+import org.apache.camel.Exchange;
+
+class Http4SpanDecorator extends HttpSpanDecorator {
+  @Nullable
+  @Override
+  protected String getHttpUrl(Exchange exchange, Endpoint endpoint) {
```

Comment:
This is mostly a copy-paste from the super-class. Can you instead introduce a new protected method into `HttpSpanDecorator`, something like `getProtocol`. Super-class will return `http`, this class can override it and return `http4`.

Even easier way is, probably, to pass this protocol prefix as constructor parameter to `HttpSpanDecorator`

Reply:
Hey good point! thanks!
Fixed